src/core_modules/capture-core/components/D2Form/D2Form.component.js

```diff
@@ -1,37 +1,23 @@
 // @flow
 import React from 'react';
-import { withStyles } from '@material-ui/core/styles';
 import log from 'loglevel';
 import { errorCreator } from 'capture-core-utils';
 import D2SectionContainer from './D2Section.container';
-import RenderFoundation from '../../metaData/RenderFoundation/RenderFoundation';
+import type { Props, PropsForPureComponent } from './D2Form.types';
 import Section from '../../metaData/RenderFoundation/Section';
-import { withAsyncHandler } from './asyncHandlerHOC';
 
-const styles = () => ({
-    container: {
-    },
+export const styles = () => ({
     containerCustomForm: {
         paddingTop: 10,
         paddingBottom: 10,
     },
 });
 
-type Props = {
-    formFoundation: RenderFoundation,
-    id: string,
-    classes: {
-        container: string,
-        containerCustomForm: string,
-    },
-    formHorizontal: boolean,
-};
-
-class D2Form extends React.PureComponent<Props> {
+class D2Form extends React.PureComponent<PropsForPureComponent> {
     name: string;
     sectionInstances: Map<string, Object>;
 
-    constructor(props: Props) {
+    constructor(props: PropsForPureComponent) {
```

Comment:
I took a moment and annotated this component since I was working with it for this PR.